core/trino-spi/src/main/java/io/trino/spi/type/MapType.java

```diff
@@ -387,33 +387,34 @@ public int relocateFlatVariableWidthOffsets(byte[] fixedSizeSlice, int fixedSize
 
     private int relocateVariableWidthData(int positionCount, int keyFixedSize, int valueFixedSize, byte[] slice, int offset)
     {
-        int writeVariableWidthOffset = positionCount / 2 * (2 + keyFixedSize + valueFixedSize);
+        int writeFixedOffset = offset;
+        int writeVariableWidthOffset = offset + (positionCount / 2 * (2 + keyFixedSize + valueFixedSize));
```

Comment:
I know it's pre-existing, but it's not clear where this formula comes from.

Also, why does it divide before multiplying? If positionCount is odd, it will round down before the multiplication happens. Is that intended?